luigi/execution_summary.py

```diff
@@ -30,36 +30,93 @@
+    *luigi.run() / luigi.build*. Here are the codes and what they mean:
+
+    =============================  =====  ========================================================== 
+    Return Code Name               Value  Meaning 
+    =============================  =====  ========================================================== 
+    SUCCESS                        0      There were no failed tasks or missing dependencies 
+    SUCCESS_WITH_RETRY             1      There were failed tasks but they all succeeded in a retry 
+    FAILED                         2      There were failed tasks
+    FAILED_AND_SCHEDULING_FAILED   3      There were failed tasks and tasks whose scheduling failed
+    SCHEDULING_FAILED              4      There were tasks whose scheduling failed
+    NOT_RUN                        5      There were tasks that were not granted run permission by the scheduler
+    MISSING_EXT                    6      There were missing external dependencies
+    =============================  =====  ==========================================================
+    
+    """
+    SUCCESS = 0
+    SUCCESS_WITH_RETRY = 1
+    FAILED = 2
+    FAILED_AND_SCHEDULING_FAILED = 3
+    SCHEDULING_FAILED = 4
+    NOT_RUN = 5
+    MISSING_EXT = 6
+
+    # Make sure all the retcodes have a (smiley, reason) in this dictionary
+    _DETAILS = {
+        SUCCESS: (":)", "there were no failed tasks or missing dependencies"),
+        SUCCESS_WITH_RETRY: (":)", "there were failed tasks but they all succeeded in a retry"),
+        FAILED: (":(" ,"there were failed tasks"),
+        FAILED_AND_SCHEDULING_FAILED: (":(", "there were failed tasks and tasks whose scheduling failed"),
+        SCHEDULING_FAILED: (":(", "there were tasks whose scheduling failed"),
+        NOT_RUN: (":|", "there were tasks that were not granted run permission by the scheduler"),
+        MISSING_EXT: (":|", "there were missing external dependencies")
+    }
 
 
 class LuigiRunResult:
     """
     Result of the execution (build/run) will be of type LuigiRunResult instead of
-    the regular Boolean response if [execution_summary]legacy_run_result is set
-    to False in the config.
+    the regular Boolean response if the keyword argument ``detailed_summary=True`` is passed to
+    build/run. A response of type **LuigiRunResult** has the following attributes:
+
+    Attributes:
+        * **summary_text_one_line** - One line summary of the progress.
+        * **summary_text** - Detailed summary of the progress.
+        * **status** - Integer Return Code. See \
+        `LuigiRetCodes <https://luigi.readthedocs.io/en/latest/api/luigi.execution_summary.html#luigi.execution_summary.LuigiRetCodes>`_ \
+        for what these codes mean.
+        * **worker** - Worker object.
+        * **execution_succeeded** - Boolean which is *True* if finally, there were no failed tasks.
+        * **scheduling_succeeded** - Boolean which is *True* if all the tasks were scheduled without errors.
```

Comment:
Is this the standard way in python to document attributes?

Reply:
I found [this link](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for docstrings in attributes. It looked better with the bullet list and **emphasis**. Since luigi hasn't documented any attributes before, I thought I could go wild with it! 🐯 
- Should I remove the emphasis and bullet-lists and make it exactly follow *the google style* ?

Reply:
> - Should I remove the emphasis and bullet-lists and make it exactly follow the google style ?

I removed the emphasis but had to keep them as bullet lists as they were all rendering in the same line. Another option was using line breaks using [pipes](https://stackoverflow.com/questions/51198270/how-do-i-create-a-new-line-with-restructuredtext) but i like bullet lists better 😅